pylint/checkers/base/basic_checker.py

```diff
@@ -452,7 +452,7 @@ def visit_expr(self, node: nodes.Expr) -> None:
             return
 
         # Warn W0133 for exceptions that are used as statements
-        if isinstance(expr, nodes.Call):
+        if isinstance(expr, nodes.Call) and isinstance(expr.func, nodes.Name):
```

Comment:
I'm having second thoughts about this additional filter.  Perhaps it does improve performance.. but perhaps there are legitimate situations where a `Call` node is _not_ a named function can be inferred to be an `Exception` type here.  That situation could be worth identifying.  Could test coverage be added for it?  What would an example be?

Reply:
From discussion in the linked issue: `Attribute` is another valid case here, in addition to `Name`.

Those two classes correspond to `ast.Name` and `ast.Attribute` in Python, and, according to the [current upstream Py3 docs for `ast.Call`](https://docs.python.org/3/library/ast.html#ast.Call):

> `func` is the function, which will often be a [Name](https://docs.python.org/3/library/ast.html#ast.Name) or [Attribute](https://docs.python.org/3/library/ast.html#ast.Attribute) object.

So it's not a strong guarantee, exactly, but handling both `Name` and `Attribute` would catch most cases.

My guess at the moment, however, is that this change _would not_ fix any performance problems -- because it's adding additional boolean checks (a performance penalty) and in most cases those will evaluate to `True` (again, referencing the Py3 docs) - so we won't be able to skip the inference step -- where we think the cost is -- in many cases.